pandas/core/indexing.py

```diff
@@ -201,6 +202,21 @@ def _get_setitem_indexer(self, key):
     def __setitem__(self, key, value):
         if isinstance(key, tuple):
             key = tuple(com.apply_if_callable(x, self.obj) for x in key)
+            # if `key` is an index to multiple columns by name, add each column
+            # if it does not already exist
```

Comment:
this might be better to do in _get_setitem_indexer

Reply:
done